src/main/java/seedu/duke/command/ListNoteCommand.java

```diff
@@ -85,61 +87,70 @@ public ListNoteCommand(Boolean isAscendingOrder, ArrayList<String> tags) {
+                    noteString.append(i + 1).append(".")
+                            .append(notebook.getNotes().get(i).getTitle())
+                            .append(InterfaceManager.LS);
                 }
             } else {
-                noteString = getSortedString(noteString, sortedNotes);
-            }
-        } else {
-            Map<Tag, ArrayList<Note>> tag = tagManager.getTagMap();
-
-            // Based on user inputted tags, will store the respective values in an ArrayList
-            // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
-            //      1 for the values corresponding to CS2113 and the other for important tag
-            List<ArrayList<Note>> values = tags.stream()
-                    .map(tag::get)
-                    .collect(Collectors.toList());
-
-            for (int i = 0; i < values.size(); i++) {
-                for (int j = 0; j < values.get(i).size(); j++) {
-                    // Account for duplicates.
-                    // In case an item has both CS2113 and Important tag
-                    if (!notes.contains(values.get(i).get(j))) {
-                        notes.add(values.get(i).get(j));
-                    }
-                }
+                noteString = new StringBuilder(getSortedString(noteString.toString(), sortedNotes));
             }
 
-            // Sort the tagged notes
-            ArrayList<Note> sortedTaggedNotes = (ArrayList<Note>) notes.stream()
-                    .filter((s) -> s instanceof Note)
-                    .sorted((a, b) -> a.getTitle().toLowerCase().compareTo(b.getTitle().toLowerCase()))
-                    .collect(Collectors.toList());
+            return noteString.toString();
+        }
 
+        Map<Tag, ArrayList<Note>> tagMap = tagManager.getTagMap();
+        ArrayList<Tag> tagList = new ArrayList<>();
 
-            if (isAscendingOrder == null) {
-                for (int i = 0; i < notes.size(); i++) {
-                    noteString += (i + 1) + "." + notes.get(i).toString();
+        for (String tag : tags) {
+            tagList.add(tagManager.getTag(tag));
+        }
+
+        // Based on user inputted tags, will store the respective values in an ArrayList
+        // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
+        //      1 for the values corresponding to CS2113 and the other for important tag
+        List<ArrayList<Note>> values = tagList.stream()
+                .map(tagMap::get)
+                .collect(Collectors.toList());
+
+        for (ArrayList<Note> value : values) {
+            for (Note note : value) {
+                // Account for duplicates.
+                // In case an item has both CS2113 and Important tag
+                if (!notes.contains(note)) {
+                    notes.add(note);
                 }
-            } else {
-                noteString = getSortedString(noteString, sortedTaggedNotes);
+            }
+        }
```

Comment:
Can do a check for the size of notes, if it has no element can return an unsuccessful message.

Reply:
Okay, made the changes will push